modules/nf-core/pymygene/templates/pymygene.py

```diff
@@ -0,0 +1,215 @@
+#!/usr/bin/env python3
+import argparse
+import mygene
+import shlex
+import sys
+
+
+class Arguments:
+    """
+    This class contains the arguments of the module.
+    """
+    def __init__(self) -> None:
+        self.input = "$gene_list"
+        self.prefix = (
+            "$task.ext.prefix"
+            if "$task.ext.prefix" != "null"
+            else "$meta.id"
+        )
+        self.output_tsv = self.prefix + ".tsv"
+        self.output_gmt = self.prefix + ".gmt"
+        self.parse_ext_args("$task.ext.args")
+
+    def parse_ext_args(self, args_string: str) -> None:
+        """
+        It parses the extended arguments.
+        """
+        # skip when there are no extended arguments
+        if args_string == "null":
+            args_string = ""
+
+        # Parse the extended arguments
+        args_list = shlex.split(args_string)  # Split the string into a list of arguments
+        parser = argparse.ArgumentParser()
+        parser.add_argument('--species', default=None, help="Comma separated of common name of the species or taxon ids")
+        parser.add_argument('--filter_go_category', default=None, help="Comma separated list of GO categories to keep. Default: all")
+        parser.add_argument('--filter_go_evidence', default=None, help="Comma separated list of GO evidence codes to keep. Default: all")
+        args = parser.parse_args(args_list)
+
+        # Convert "null" values to default values
+        for attr in vars(args):
+            value = getattr(args, attr)
+            if value == "null":
+                setattr(args, attr, parser.get_default(attr))
+
+        # Assign args attributes to self attributes
+        for attr in vars(args):
+            setattr(self, attr, getattr(args, attr))
+
+
+class Version:
+    """
+    This class contains the functions to get the versions of the modules used in the script.
+    """
+
+    @staticmethod
+    def get_versions(modules: list) -> dict:
+        """
+        This function takes a list of modules and returns a dictionary with the versions of each module.
+        This dictionary will also contain the version of python.
+        """
+        dic = {
+            **{"python": '.'.join(map(str, sys.version_info[:3]))},
```

Comment:
I suggest not doing this. I recently had to correct a couple of R modules because the version of R had drifted for Conda. Basically, interpreter versions are not necessarily stable in Conda, so don't snapshot them.